packages/babel-generator/src/generators/methods.js

```diff
@@ -109,7 +109,8 @@ export function ArrowFunctionExpression(node: Object) {
   if (
     node.params.length === 1 &&
     t.isIdentifier(firstParam) &&
-    !hasTypes(node, firstParam)
+    !hasTypes(node, firstParam) &&
+    !node.async
   ) {
     if (
       this.format.retainLines &&
```

Comment:
We can move `node.async` to here
```
this.format.retainLines || node.async
```
so we are instructing that if `async` presents, we try to retain lines and print `(` if the function body is in different lines with the async arrow.